samples/subsys/smf/smf_calculator/src/smf_console_cmds.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023 Glenn Andrews
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include "smf_calculator_thread.h"
+#include <stdlib.h>
+
+static int cmd_smf_key(const struct shell *sh, size_t argc, char **argv)
+{
+	struct calculator_event event;
+	char c = argv[1][0];
+
+	event.operand = c;
+
+	if ((c >= '1') && (c <= '9')) {
```

Comment:
have you considered using a `switch (c)` for this one?

Reply:
Yeah, but wasn't so keen on:
```
	case '1':
	case '2':
	case '3':
	case '4':
	case '5':
	case '6':
	case '7':
	case '8':
	case '9':
		event.event_id = DIGIT_1_9;
		break;
```